network/src/connectivity_manager/mod.rs

```diff
@@ -800,6 +821,10 @@ impl Addresses {
     fn get(&self, idx: usize) -> Option<&NetworkAddress> {
         self.0.iter().flatten().nth(idx)
     }
+
+    fn union(&self) -> Vec<NetworkAddress> {
+        self.0.iter().flatten().cloned().collect()
+    }
```

Comment:
this looks like `flatten` not `union`

Reply:
Yeah, you're right.

Need a stable merge of the multiple vecs to do it.  It wasn't really being used for anything so I did it quickly, but let me fix it.

Reply:
Fixed, it now gets a unique union of the addresses.  Order is now lost, but it's not being used particularly for anything at the moment.  I think we can come back and deal with ordering and refactoring this later.